portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -1,17 +1,29 @@
 /**
-* A comment has an id, text and an author. 
+* A comment has an id, text, an author and a mood.
 */
 
 package com.google.sps.data;
 
 public final class Comment {
   public final long id;
+  public final Long timestamp;
   public final String text;
   public final String author;
+  public final String mood;
 
-  public Comment(long id, String text, String author) {
+  public Comment(long id, String text, String author, String mood, Long timestamp) {
```

Comment:
Is it necessary to use Long here?

In Java it's common to use the primitives.

Auto-boxing will take care of cases where you really need a Boolean, Long, Integer etc.
https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html

I see that the datastore entity takes objects. I guess you could still use primitives everywhere but when dealing with the datastore entity.

Reply:
I am using the object Long because in Comparator interface long and int cannot be dereferenced.

Reply:
Can you just use `Comparator.comparingLong` and `Comparator.comparingInt`?

https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#comparingLong-java.util.function.ToLongFunction-

Something along the lines of

```
Comparator<Long> byTimestamp = Comparator.comparingLong(Comment::getTimestamp);
```

should work.

Alternatively, when writing a comparator for `long` or `int` yourself, you could just return the difference between the two, using subtraction rather than compareTo method.

Reply:
Okay, I will use them.